major-mode-hydra.el

```diff
@@ -134,10 +134,14 @@ exactly the same structure as that in `pretty-hydra-define' or
 
 (defun major-mode-hydra-dispatch (mode)
   "Summon the hydra for given MODE (if there is one)."
-  (let ((hydra (major-mode-hydra--body-name-for mode)))
-    (if (fboundp hydra)
-        (call-interactively hydra)
-      (message "Major mode hydra not found for %s" mode))))
+  (catch 'done
+    (while mode
+      (let ((hydra (major-mode-hydra--body-name-for mode)))
+        (when (fboundp hydra)
+          (call-interactively hydra)
+          (throw 'done t)))
+      (setq mode (get-mode-local-parent mode)))
+    (user-error "Major mode hydra not found for %s" mode)))
```

Comment:
Do this log the ancestor mode name rather than the original `mode` passed to the function? If that's the case it might be a bit confusing? I'd suggest a message like `Major mode hydra not found for %s or its parent modes` and put the original `mode` in it.

Reply:
Thanks, I've fixed that.